cluster/router/url_utils.go

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package router
+
+import (
+	"strings"
+)
+
+import (
+	"github.com/apache/dubbo-go/common"
+)
+
+func isMatchGlobPattern(pattern string, value string, param *common.URL) bool {
+	if param != nil && strings.HasPrefix(pattern, "$") {
+		pattern = param.GetRawParam(pattern[1:])
+	}
+	return isMatchInternalPattern(pattern, value)
+}
+
+func isMatchInternalPattern(pattern string, value string) bool {
```

Comment:
How about declaring an interface like
``` go
type Matcher interface {
   match(pattern, value) bool
}
``` 
and make this method as default implementation?
Or, rename "url_utils" to "match_utils"?

Reply:
i will rename the file ,but i think no need to declare an interface. Because it just for router module ,and the pattern no more else implement.